### DISCUSSION TITLE
Enable list_prompts/search_prompts using client

### DISCUSSION THREAD

**Discussion by krlng** [langfuse/langfuse#2797] [Ideas]
### Describe the feature or potential improvement

It would be great to list all available prompt-names with the option to filter by tag or label. 

Our concrete use-case is a multi-agent-setup, where one agents wants to load the descriptions for all other agents. This would be a good use-case, how ever I imagine there are probably many more.

**Comment by marcklingen** [comment]
Thank you for bringing this up. This functionality is available through the API, and I agree that including a wrapper in the SDK clients would be useful.

Endpoint ([reference](https://api.reference.langfuse.com/#get-/api/public/v2/prompts)): `GET /api/public/v2/prompts`

**Comment by larsklei** [comment]
If you need an ad-hoc solution and use the python sdk, you might do the following:
```
from langfuse import Langfuse

lf_client = Langfuse()
lf_api_wrapper = lf_client.client
prompts = lf_api_wrapper.prompts.list(
label="your_label"
tag="your_tag"
)
```

It seems that the FernLangfuse Class is the actual API wrapper in the python sdk.

**Reply by marcklingen** [reply]
yes this works, however we plan to add interfaces that are then also covered by tests and semver. The `FernLangfuse` class is auto-generated and currently not covered by tests as a rather internal interface. As we do not plan to remove it, it is safe to use for now though

**Reply by krlng** [reply]
that sounds great. We will use this for now, thank you.

**Reply by sapirm-eleos** [reply]
Hey, pinging this issue, any updates on this being covered by tests?
We'd like to use this functionality for filtering prompts but it sounds like it's not ready for production use yet, is it?

**Reply by hassiebp** [reply]
@sapirm-eleos - you can use 

```python
from langfuse import get_client

langfuse = get_client()

langfuse.api.prompts.list(*)
```

to list prompts given filters.
